plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -301,14 +344,74 @@ func (cache *schedulerCache) RemovePod(pod *v1.Pod) error {
 	return nil
 }
 
+func (cache *schedulerCache) AddResourceClass(rClass *v1.ResourceClass) error {
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+	var err error
+	//fmt.Println(file_line())
+	fmt.Printf("%s schedulerCache: %p,  NumNodes=> %d\n", file_line(), cache, len(cache.nodes))
+	_, ok := cache.resourceClasses[rClass.Name]
+	if !ok {
+		rc := &ResourceClassInfo{}
+		cache.resourceClasses[rClass.Name] = rc
+		rc.resClass = rClass
+		for _, info := range cache.nodes {
+			fmt.Printf("\n%s cache.nodes %+v\n", file_line(), cache.nodes)
+			rcPerNodeInfo, err := info.AddResourceClass(rClass, info.node)
```

Comment:
This smells a little too much like hard coupling between nodes and resource classes. Or am I missing something non-obvious here?